Esp32

Misca1234 opened the pull request:
Push ESP 32
https://www.espressif.com/sites/default/files/documentation/esp32_hardware_design_guidelines_en.pdf

It comes in two flavours, a 5x5 QFN and a 6x6 QFN

The symbol is exactly the same for both of them
ESP32-D0WDQ6    6x6
ESP32-D0WD, alias ESP32-D2WD, ESP32-S0WD  5x5

![bild](https://user-images.githubusercontent.com/25547797/42178890-43c21554-7e32-11e8-8b06-82bfa7055013.png)

![bild](https://user-images.githubusercontent.com/25547797/42178901-4b0be38a-7e32-11e8-932d-8394b1adbd51.png)

![bild](https://user-images.githubusercontent.com/25547797/42178951-7d49810e-7e32-11e8-8190-579d348d1ee2.png)

![bild](https://user-images.githubusercontent.com/25547797/42178939-71c2f298-7e32-11e8-885d-f3501da7f565.png)


![bild](https://user-images.githubusercontent.com/25547797/42178908-5662bb00-7e32-11e8-970e-729ae83fa7d0.png)

![bild](https://user-images.githubusercontent.com/25547797/42178916-5e0b1848-7e32-11e8-926a-d60ae419515a.png)

![bild](https://user-images.githubusercontent.com/25547797/42178927-65b0c14c-7e32-11e8-8ce0-f7390d4aadfa.png)

![bild](https://user-images.githubusercontent.com/25547797/42178820-08fae360-7e32-11e8-9e07-fe09c94f41af.png)

![bild](https://user-images.githubusercontent.com/25547797/42179074-fa2d65b4-7e32-11e8-8fd2-fe2083eed0dc.png)

![bild](https://user-images.githubusercontent.com/25547797/42179086-02896fb4-7e33-11e8-8a46-4d5e4509ec09.png)

![bild](https://user-images.githubusercontent.com/25547797/42179094-08b0c5c2-7e33-11e8-9176-4c5d0f8e85d0.png)









------------
Thanks for creating a pull request to contribute to the KiCad libraries! To speed up integration of your PR, please check the following items:

- [ ] Provide a URL to a datasheet for the symbol(s) you are contributing
- [ ] An example screenshot image is very helpful
- [ ] Ensure that the associated footprints match the [official footprint library](https://github.com/kicad/kicad-footprints)
- [ ] If there are matching footprint PRs, provide link(s) as appropriate
- [ ] Check the output of the Travis automated check scripts - fix any errors as required

poeschlr commented:
The datasheet you link is for a module not really for a qfn version. Is there a better datasheet available?

Misca1234 commented:
This one should be better, I change in the symbols
https://www.espressif.com/sites/default/files/documentation/esp32_datasheet_en.pdf

myfreescalewebpage commented:
Hi @Misca1234 , thanks for contributing,

A few comments I have during my review:

ESP32-D0WD:
- [ ] You can complete the description: `Single 2.4 GHz Wi-Fi-and-Bluetooth combo chip, dual core, no internal flash, QFN-48` according to datasheet page 46
- [ ] Footprint filter should be `QFN*1EP*5x5mm*P0.35mm*` (no EP size in it)
- [ ] VDD* input pins should not be stacked
- [ ] "VDD_SDIO" should be moved to the top of the symbol. Datasheet indicates a link with "VDD3P3_RTC", so I propose to get these two pins together on the right of the VDD* group
- [ ] Name of pin 49 is "GND" only
- [ ] I clearly do not understand how you have chosen to group the pins except for SD/Crystal/CAP/Sensor. Can you detail your logic ? Why MTCK not with MTDI etc for example.

ESP32-D0WDQ6:
- [ ] Footprint filter should be `QFN*1EP*6x6mm*P0.4mm*`
- [ ] VDD* input pins should not be stacked
- [ ] "VDD_SDIO" should be moved to the top of the symbol. Datasheet indicates a link with "VDD3P3_RTC", so I propose to get these two pins together on the right of the VDD* group
- [ ] Name of pin 49 is "GND" only
- [ ] I clearly do not understand how you have chosen to group the pins except for SD/Crystal/CAP/Sensor. Can you detail your logic ? Why MTCK not with MTDI etc for example.

ESP8089:
- [ ] Datasheet link is broken, I have found only "http://gamma.spb.ru/images/pdf/esp8089_datasheet_en.pdf", have you got a better one ?
- [ ] Footprint should be `Package_DFN_QFN:QFN-32-1EP_5x5mm_P0.5mm_EP3.7x3.7mm` according to datasheet page 18
- [ ] Footprint filter should be `QFN*1EP*5x5mm*P0.5mm*`
- [ ] VDD* input pins should not be stacked
- [ ] Name of pin 7 is `~CHIP_PU`
- [ ] Pin 24 id "GPIO5", it is not a Power supply pin
- [ ] Name of pin 32 should be `~EXT_RST` to get the over-bar (B indicate active low)
- [ ] Name of pin 33 is "GND" only
- [ ] Type of pin "RES12K" should be Passive to avoid ERC errors
- [ ] Type of pin 29 "VDDD" is Power Input

Cheers,
Joel

Misca1234 commented:
ESP32-D0WD:

> You can complete the description: Single 2.4 GHz Wi-Fi-and-Bluetooth combo chip, dual core, no internal flash, QFN-48 according to datasheet page 46

Changed

> Footprint filter should be QFN*1EP*5x5mm*P0.35mm* (no EP size in it)

Changed


> VDD* input pins should not be stacked

They are seperated


> "VDD_SDIO" should be moved to the top of the symbol. 

It is placed as in accordance to a normal power regulator output


> Name of pin 49 is "GND" only

Changed

> I clearly do not understand how you have chosen to group the pins 
MTDI, GPIO0, GPIO2, MTDO, GPIO5
is grouped as strapped pins, page 11, 
there are other ways to group, such as JTAG etc.

SD_CMD, SD_CLK, SD_DATA_0, SD_DATA_1, SD_DATA_2
Is grouped for SPI, page 30

MTCK, MTMS, there was room for them there


ESP32-D0WDQ6:

> Footprint filter should be QFN*1EP*6x6mm*P0.4mm*

Changed

> VDD* input pins should not be stacked

Changed

> "VDD_SDIO" should be moved to the top of the symbol. 

It is placed as in accordance to a normal power regulator output

> Name of pin 49 is "GND" only

Changed

> I clearly do not understand how you 

As before


ESP8089:

> Datasheet link is broken

We can try this one
https://www.espressif.com/sites/default/files/documentation/0a-esp8285_datasheet_en.pdf

> Footprint should be Package_DFN_QFN:QFN-32-1EP_5x5mm_P0.5mm_EP3.7x3.7mm according to datasheet page 18

This does not exist, keeping current one or we should add the 3.7x3.7

> Footprint filter should be QFN*1EP*5x5mm*P0.5mm*

Changed

> VDD* input pins should not be stacked

Changed

> Name of pin 7 is ~CHIP_PU

We are not reading same data sheet, mine says this

![bild](https://user-images.githubusercontent.com/25547797/51440389-d12cfe80-1cc6-11e9-820e-c75631a89f8a.png)


> Pin 24 id "GPIO5", it is not a Power supply pin

We are not reading same data sheet, mine says this

![bild](https://user-images.githubusercontent.com/25547797/51440396-dc802a00-1cc6-11e9-9608-628dbe221ec9.png)

> Name of pin 32 should be ~EXT_RST to get the over-bar (B indicate active low)

Tilde added to beginning of string to indicate active low


> Name of pin 33 is "GND" only


> Type of pin "RES12K" should be Passive to avoid ERC errors

It is an input according to the data sheet, If there is a bug in 
ERF checker then ERC checker should be fixed instead

> Type of pin 29 "VDDD" is Power Input

Changed

myfreescalewebpage commented:
Hello @Misca1234 

> VDD* input pins should not be stacked
They are seperated

They are not all separated today, you should.

> "VDD_SDIO" should be moved to the top of the symbol.
It is placed as in accordance to a normal power regulator output

Regulator output on the right is an accepted KLC exception. Other MCU have "VDD CORE" power output, always placed on the top. So this one should complies with the KLC too.

> Datasheet link is broken
We can try this one
https://www.espressif.com/sites/default/files/documentation/0a-esp8285_datasheet_en.pdf

I don't think it applies to ESP8089. It is ESP8285 only.

> Footprint should be Package_DFN_QFN:QFN-32-1EP_5x5mm_P0.5mm_EP3.7x3.7mm according to datasheet page 18
This does not exist, keeping current one or we should add the 3.7x3.7

You should create it with the scripts and make a PR, it is not allowed to round this.

> Name of pin 7 is ~CHIP_PU
We are not reading same data sheet, mine says this

I will be happy to remove my comment if you provide the right datasheet :)

> Name of pin 32 should be ~EXT_RST to get the over-bar (B indicate active low)
Tilde added to beginning of string to indicate active low

You should also remove the "B"

> Type of pin "RES12K" should be Passive to avoid ERC errors
It is an input according to the data sheet, If there is a bug in
ERF checker then ERC checker should be fixed instead

The ERC check has limitations and it is not possible to handle every thing. We use Passive for this kind of pins in several hundred of devices (and maybe more!).

Joel

myfreescalewebpage commented:
@Misca1234 any news here ? Thanks !

Misca1234 commented:
> any news here
No, not really

myfreescalewebpage commented:
No news from the original author, indicate it is abandoned.
